src/commands/deleteVirtualMachine/getResourcesAssociatedToVm.ts

```diff
@@ -0,0 +1,96 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ComputeManagementClient } from "@azure/arm-compute";
+import { NetworkManagementClient, NetworkManagementModels } from "@azure/arm-network";
+import { localize } from "../../localize";
+import { VirtualMachineTreeItem } from "../../tree/VirtualMachineTreeItem";
+import { createComputeClient, createNetworkClient } from "../../utils/azureClients";
+import { getNameFromId, getResourceGroupFromId } from "../../utils/azureUtils";
+import { networkInterfaceLabel, ResourceToDelete, virtualNetworkLabel } from "./deleteConstants";
+
+export async function getResourcesAssociatedToVm(node: VirtualMachineTreeItem): Promise<ResourceToDelete[]> {
+
+    const associatedResources: ResourceToDelete[] = [];
+
+    const networkNames: string[] = [];
+    if (node.data.networkProfile?.networkInterfaces) {
+        for (const networkRef of node.data.networkProfile?.networkInterfaces) {
+            if (networkRef.id) {
+                networkNames.push(getNameFromId(networkRef.id));
```

Comment:
Honestly, I don't think we want to use `resourceGroupName` at all in this file and seems like networks is the only place you still use it. I'd just change `networkNames` to `networkIds` and do `getNameFromId` and `getResourceGroupFromId` below.